src/Chapter2.hs

```diff
@@ -336,7 +348,11 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList lhs rhs xs
+    | lhs < 0 || rhs < 0 || lhs > rhs = []
+    | otherwise = take takeLength (drop lhs xs)
+        where
+            takeLength = min (length xs) (rhs - lhs + 1)
```

Comment:
You don't need to worry about it, `take` will take what it can, so no need to check the length (and also, `take` will work with the dropped list, not the xs, so the real length of the list it would work with would be smaller).
`take` is the lazy function, so it can even work with infinite lists, it will go element-by-element until it takes enough. or the list is empty 👍🏼